frontend/src/Editor/LeftSidebar/index.js

```diff
@@ -12,31 +12,39 @@ import { LeftSidebarGlobalSettings } from './SidebarGlobalSettings';
 import { ConfirmDialog } from '@/_components';
 import config from 'config';
 
-export const LeftSidebar = ({
-  appId,
-  switchDarkMode,
-  darkMode = false,
-  components,
-  toggleComments,
-  dataSources = [],
-  dataSourcesChanged,
-  dataQueriesChanged,
-  errorLogs,
-  appVersionsId,
-  globalSettingsChanged,
-  globalSettings,
-  currentState,
-  appDefinition,
-  setSelectedComponent,
-  removeComponent,
-  runQuery,
-  toggleAppMaintenance,
-  is_maintenance_on,
-  isSaving,
-  isUnsavedQueriesAvailable,
-}) => {
+export const LeftSidebar = forwardRef((props, ref) => {
   const router = useRouter();
+  const {
+    appId,
+    switchDarkMode,
+    darkMode = false,
+    components,
+    toggleComments,
+    dataSources = [],
+    dataSourcesChanged,
+    dataQueriesChanged,
+    errorLogs,
+    appVersionsId,
+    globalSettingsChanged,
+    globalSettings,
+    currentState,
+    appDefinition,
+    setSelectedComponent,
+    removeComponent,
+    runQuery,
+    toggleAppMaintenance,
+    is_maintenance_on,
+    isSaving,
+    isUnsavedQueriesAvailable,
+  } = props;
   const [showLeaveDialog, setShowLeaveDialog] = useState(false);
+  const [showDataSourceManagerModal, toggleDataSourceManagerModal] = React.useState(false);
```

Comment:
Please replace `React.useState` with `useState` as we're importing it at the top and using it in the file